### DISCUSSION TITLE
Need help setting up MK3S+ with Continous Print

### DISCUSSION THREAD

**Discussion by rogiepoo** [Pierro55/Loop#7] [Q&A] [answered]
Hi! I am on the verge of total success in setting up my MK3S+ with Octo's Continuous Print. Also I have the 3DQue VAAPR bed overlay which I have several months' experience with.  My start code seems to work great, and I am able to write my own bed clearing code (uses the nozzle to knock of a part).  However, what is happening is that after the first print finishes, after going home (correctly) it immediately goes into mesh bed leveling, skipping the cooling cycle. At this point, the print is still on the bed, yikes! Since my test print is very small, it is not crashed into by mesh bed leveling, and so it completes it and goes home. THEN the cooling cycle begins. [I have it set up so that when Y =0, a momentary toggle switch mounted at the back of the printer is bumped into. When depressed, this relay switch activates a low-level current which in a relay circuit, which in turn activates 115V current from a wall outlet to activate a powerful cooling fan. This works great; the bed & prints are cooled in five minutes. I can and will share setup info] Then it does mesh bed leveling again TWICE more, before moving onto the bed clearing script. It runs the clearing script, and then....drum roll please...it's time for mesh bed leveling! With an ice cold nozzle and bed, of course. When done, it homes with a Y=0 and engages the fan switch, while simultaneously attempting to heat the bed & nozzle.  I have modified the start gcode (kept in Prusaslicer) to run the nozzle and heating sequence twice: once with the noz to 170C and bed to 60C, followed by a mesh bed leveling sequence, and then another heating cycle with noz to 240 and bed 60 (so no change to bed temp, just stays 60). 240 kind of high but works great for PLA I am using (PolyTerra Matte) and a VAAPR bed.  This two step approach help prevents unwanted PLA buildup on the nozzle before entering the purge sequence followed by the actual printing of the part.  So in summary, the start code works good, the clearing script is good, but I have a surplus of mesh bed levelings taking place, and I need to delete those and just get on with bed cooling followed by the clearing routine, then just go to the next print (which has the heating1, leveling, heating2 as start code).  Where is all this mesh leveling coming from? Is this in the firmware for the printer?  How do I get rid of it?  Thanks for any and all suggestions. R

**Comment by rogiepoo** [comment]
Forgot to add my system info. 
[octoprint-systeminfo-20230226103653.zip](https://github.com/Pierro55/Loop/files/10834435/octoprint-systeminfo-20230226103653.zip)

**Reply by Pierro55** [reply]
Hi rogiepoo,
Do you have the clearing script in the continuous print plugin or in the slicer? 
I suggest keeping it in the slicer for simplicity and for being able to make changes occasionally without the need to keep track of two different programs.

You can find my clearing script at the bottom of the following page (you’ll probably have to change some bits to fit your application): https://github.com/Pierro55/Loop/tree/main/Instructions

A suggestion on how to move forward:

Put your clearing script in your slicer of choice and print without Octoprint.
Open the sliced Gcode and copy it. Paste it at the end of the Gcode and therefore looping it one time.
Watch what happens after the first run is finished.
If it prints just fine, there is something wrong with your Octoprint setup.

**Comment by rogiepoo** [comment]
Pierro55,
1) Thank you! !!
2) I believe with the "loop" idea, you are suggesting to place an entire
test print within the end gcode itself in order to create a pseudo
continuous print 2x queue, correct?
3) This morning I removed the clearing script from the Continuous Events
and also the scrips, and placed it into the slicer (Prusaslicer). I am
still getting a mesh leveling that fires before bed clearing. Therefore I
am quite sure this is an octoprint setup issue.
4) I may try uninstalling the continuous print plug-in and then performing
a print with the cooling and clearing in the slicer end gcode to see if I
can get a "cooling and clearing event" without a preceding "mesh leveling
event".  That would be persuasive in determining whether the issue rests
with Octoprint versus Cont Print.
5) If I get more confused, I will try the loop as you suggest (provided I
understand that correctly).
6) Thank you for trying to help me!!!!

*RogRoger Rasmussen*
Coastmans Scenic Products
(360) 820-3553
17 Hamlet
Port Orford, Oregon 97465
website: www.coastmans.com
modelmaking: www.coastalmodelworks.com
email: ***@***.***

**Comment by rogiepoo** [comment]
I did uninstall continuous print and I do still have the issue of mesh
leveling preceding cooldown, even w/o continuous print plugin installed!
So there is definitely something wrong w/my octo setup. But I have not
tried printing from SD card yet to rule out that it's not the machine
itself.
r

*RogRoger Rasmussen*
Coastmans Scenic Products
(360) 820-3553
17 Hamlet
Port Orford, Oregon 97465
website: www.coastmans.com
modelmaking: www.coastalmodelworks.com
email: ***@***.***

**Reply by Pierro55** [reply]
No problem, glad if I am able to help :)

2) yes, that is exactly what I mean

Glad to hear that you could somewhat pinpoint the problem.
If you want to, I can take a look at your clearing and cooling script. Would probably be best if you send a complete Gcode including clearing and cooling parts.

**Comment by rogiepoo** [comment]
Hi,
Sorry not to drag on here, but I am just so excited to make headway on
this. I've been working on solving this all weekend.
Your suggestion was key.  After uninstalling ContPrint, and running the
file from an SD card, I STILL had the issue!
So I have been using Prusaslicer. So I switched to CURA. I sliced it, but I
did not put in the end gcode that I had in Prusaslicer. That introduces a
double variable, but
I wanted to rule out the issue was not in the PrusaMKS+ firmware itself. I
then printed from SD the Cura sliced file, and did NOT have the issue of
repetitive mesh leveling.
I see a couple ways for me to move forward: I can add the bed clearing
script to end g-code in CURA, slice that, and see if I do/do not acquire
repetitive leveling.
And /or in a separate experiment, remove the start and end g-code from the
Prusaslicer and see if that remediates repetitive mesh bed leveling.
So I guess I'm not sure if adding the bed clearing g-code to the endcode in
Prusaslicer is causing repetitive mesh bed leveling.
Also I disconnected RPi.
I'm obviously way a noob here and I am approaching diagnosis very manually.
I am next going to try to open the g-codes from the slicer(s) and review
them. That's a first for me, to even do that.
What is the preferred way of sending scripts?  I could copy & paste them
here, or attach Notepad++ files.
--> OK I attached Notepad ++'s.
Please note that I have been experimenting using these in different
combinations and locations (ContinPrint vs Prusaslicer).
Thank you again,

*RogRoger Rasmussen*
Coastmans Scenic Products
(360) 820-3553
17 Hamlet
Port Orford, Oregon 97465
website: www.coastmans.com
modelmaking: www.coastalmodelworks.com
email: ***@***.***


On Mon, Feb 27, 2023 at 11:07 AM Pierro55 ***@***.***> wrote:

> No problem, glad if I am able to help :)
>
>    1. yes, that is exactly what I mean
>
> Glad to hear that you could somewhat pinpoint the problem.
> If you want to, I can take a look at your clearing and cooling script.
> Would probably be best if you send a complete Gcode including clearing and
> cooling parts.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Pierro55/Loop/discussions/7#discussioncomment-5127102>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A56POOVZGWRMO6RJYAE5MADWZT3O5ANCNFSM6AAAAAAVISTNUQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

M104 T0 S0 ; extruder heater off
M140 S0 ; heated bed heater off
G28 ; home all axes
G90 ; absolute positioning
G0 X25 Y0 Z20 F6000 ; raise the extruder 2cm go to COOLING location at 6000% normal speed
M190 R27 ; wait for bed to cool to 27C
G28 ; home all axes
G90 ; absolute positioning
G0 X25 Y5 Z20 F6000 ; raise the extruder 2cm go to starting point at 3000% normal speed
G0 X245 Y5 Z20 F6000 ; go to lower right corner of bed, hold Z at 20
G0 X245 Y275 Z20 F6000 ; go to upper right of bed, holding Z at 20
G0 X215 Y275 Z20 F6000 ; go over to the left by 3 cm
G0 X215 Y275 Z1 F6000 ; drop down the nozzle to 1mm over the bed
G0 X215 Y5 Z1 F6000 ; slowly complete first wipe vector
G0 X215 Y5 Z20 F6000: lift Z 20mm up @ 3000%speed
G0 X215 Y275 Z20 F6000 ; go to top of bed, holding Z at 20
G0 X185 Y275 Z20 F6000 ; go over to the left by 3 cm
G0 X185 Y275 Z1 F6000 ; drop down the nozzle to 1mm over the bed
G0 X185 Y5 Z1 F6000 ; slowly complete 2nd wipe vector
G0 X185 Y5 Z20 F6000: lift Z 20mm up @ 3000%speed
G0 X185 Y275 Z20 F6000 ; go to top of bed, holding Z at 20
G0 X155 Y275 Z20 F6000 ; go over to the left by 3 cm
G0 X155 Y275 Z1 F6000 ; drop down the nozzle to 1mm over the bed
G0 X155 Y5 Z1 F6000 ; slowly complete 3rd wipe vector
G0 X155 Y5 Z20 F6000: lift Z 20mm up @ 3000%speed
G0 X155 Y275 Z20 F6000 ; go to top of bed, holding Z at 20
G0 X125 Y275 Z20 F6000 ; go over to the left by 3 cm
G0 X125 Y275 Z1 F3000 ; drop down the nozzle to 1mm over the bed
G0 X125 Y5 Z1 F6000 ; slowly complete 4th wipe vector
G0 X125 Y5 Z20 F6000: lift Z 20mm up @ 3000%speed
G0 X125 Y275 Z20 F6000 ; go to top of bed, holding Z at 20
G0 X95 Y275 Z20 F6000 ; go over to the left by 3 cm
G0 X95 Y275 Z1 F6000 ; drop down the nozzle to 1mm over the bed
G0 X95 Y5 Z1 F6000 ; slowly complete 5th wipe vector
G0 X95 Y5 Z20 F6000: lift Z 20mm up @ 3000%speed
G0 X95 Y275 Z20 F6000 ; go to top of bed, holding Z at 20
G0 X65 Y275 Z20 F6000 ; go over to the left by 3 cm
G0 X65 Y275 Z1 F6000 ; drop down the nozzle to 1mm over the bed
G0 X65 Y5 Z1 F6000 ; slowly complete 6th wipe vector
G0 X65 Y5 Z20 F6000: lift Z 20mm up @ 3000%speed
G0 X65 Y275 Z20 F6000 ; go to top of bed, holding Z at 20
G0 X35 Y275 Z20 F6000 ; go over to the left by 3 cm
G0 X35 Y275 Z1 F6000 ; drop down the nozzle to 1mm over the bed
G0 X35 Y5 Z1 F6000 ; slowly complete 6th wipe vector
G0 X35 Y5 Z20 F6000: lift Z 20mm up @ 3000%speed
G0 X35 Y275 Z20 F6000 ; go to top of bed, holding Z at 20
G0 X5 Y275 Z20 F6000 ; go over to the left by 3 cm
G0 X5 Y275 Z1 F6000 ; drop down the nozzle to 1mm over the bed
G0 X5 Y5 Z1 F6000 ; slowly complete 6th wipe vector
G28 ; home all axes

; Rogers Start Code for Prusa-Octo_ContinPrint:
; Original Prusa i3 MK3S & MK3S+ 0.25 nozzle
; Start G-code is as follows:

M862.3 P "[printer_model]" ; printer model check
M862.1 P[nozzle_diameter] ; nozzle diameter check
M115 U3.11.0 ; tell printer latest fw version
G90 ; use absolute coordinates
M83 ; extruder relative mode
M104 S170 ; set extruder temp
M140 S[first_layer_bed_temperature] ; set bed temp
M190 S[first_layer_bed_temperature] ; wait for bed temp
M109 S170 ; wait for extruder temp
G0 X0 Y1 Z1.5 F3000 ; home all without mesh bed level
G80 ; mesh bed leveling
G0 X0 Y1 Z1.5 F3000 ; home all without mesh bed level
G1 Z2 F720 ; Not sure exactly why this line. It was in the Prusaslicer start code I had and the entry was 0.2 but I changed it to 2 so I could better see & monitor the nozzle dribbles.
G90 ; Use absolute coordinates
M83 ; Extruder relative mode
G92 E0
M104 S[first_layer_temperature] ; Set extruder temp
M140 S[first_layer_bed_temperature] ; Set bed temp
M190 S[first_layer_bed_temperature] ; Wait for bed temp
M109 S[first_layer_temperature] ; Wait for extruder temp
G92 E0
G1 X200 Y10 Z1.5 F3000; Go to RIGHT of bed to begin purge line
G1 X225 E20 F400; Draw purge line
G1 E-0.8; Retract
G0 X225 Y50 F4000; Move quickly away
G92 E0
M221 S{if layer_height<0.075}100{else}95{endif}
; Don't change E value below. Excessive value can damage the printer.
{if ***@***.*** nozzle MK3.*/}M907 E430 ; set extruder motor current{endif}


; Note the dual heating scripts are intended to reduce nozzle dribble prior to the purge line. Heats to 170C, then mesh leveling, then heats to 240(for Polyterra matte) or 220 (for Polylite), determined in slicer. 



G28
M190 R27 ; bed cooldown to 27C
{if max_layer_z < max_print_height}G1 Z{z_offset+min(max_layer_z+1, max_print_height)} F720 ; Move print head up{endif}
G1 X0 Y200 F3600 ; park
{if max_layer_z < max_print_height}G1 Z{z_offset+min(max_layer_z+49, max_print_height)} F720 ; Move print head further up{endif}
G1 X0 Y0 Z1.5 F3600 ; park at origin
M221 S100 ; reset flow
M900 K0 ; reset LA
{if print_settings_id=~/.*(DETAIL @MK3|QUALITY @***@***.*** nozzle MK3).*/}M907 E538 ; reset extruder motor current{endif}
M104 S0 ; turn off temperature
M140 S0 ; turn off heatbed

**Reply by Pierro55** [reply]
Thanks for your code. I will dig through it tomorrow after work.

Your manual diagnosis approach btw is very good. This is actually normal in a scientific environment, always change one variable at a time. That gives you the best understanding of the problem your are trying to fix.

And great to hear that you pinpointed it even further! :)

Btw. what I saw at first glance in the Gcode is that you home the printer one time before the cooling pause and one time afterwards.
Does this maybe create a problem as the print is still on the bed when you home before and after the cooling pause?

**Comment by rogiepoo** [comment]
One more thing that might be useful, attached, the g-code for the test
object (very simple, 1 inch long line).
This is the one I had sliced in Prusaslicer (before trying it in Cura).
I looked at it in Notepad++ and searched for G80 (mesh bed leveling)
finding this only on two lines.
Thanks,

*RogRoger Rasmussen*
Coastmans Scenic Products
(360) 820-3553
17 Hamlet
Port Orford, Oregon 97465
website: www.coastmans.com
modelmaking: www.coastalmodelworks.com
email: ***@***.***

**Comment by rogiepoo** [comment]
I think I probably home it too frequently, I mean, there may be some
redundancy there.
Probably don't need the one after the cooling pause.
I don't have to worry too much about most of my prints. They are mostly
quite flat, within a range of 1-10 mm high.
They are to make the kits which I offer on my website.  Most of the parts
fold and/or combine in some way to make a taller structure.
I never thought this would make a programmer of me! :)

*RogRoger Rasmussen*
Coastmans Scenic Products
(360) 820-3553
17 Hamlet
Port Orford, Oregon 97465
website: www.coastmans.com
modelmaking: www.coastalmodelworks.com
email: ***@***.***

**Comment by rogiepoo** [comment, marked as answer]
My wife sleuthed this thing big time. Came down here and learned Gcodes
(enough of them) in an hour or two to figure out that G28 on the Marlin
site is not G28 in Prusaland. G28 in Prusaslicer INCLUDES MESH BED
LEVELING! You have to type G28 W to not include the mesh bed leveling. It
is finally bed-clearing w/o mesh bed leveling. This took me all day (until
she finally popped in).  I thought it was Continuous Print, Octoprint,
Prusaslicer, the machine, man, I was lost!
Let me know if you want the specs on my cooling setup! 5 minutes, I promise!
Thanks again,

*RogRoger Rasmussen*
Coastmans Scenic Products
(360) 820-3553
17 Hamlet
Port Orford, Oregon 97465
website: www.coastmans.com
modelmaking: www.coastalmodelworks.com
email: ***@***.***

**Reply by Pierro55** [reply]
That is great to hear!! 

I would definitely be interested in the specs of your cooling setup :)

**Comment by rogiepoo** [comment]
Everything seems to be working perfectly. Now I'd like to improve my purge
line and make it like yours; several passes back and forth (layers) will
build it up and make it easier to knock off with the nozzle tip.
Have you found your arm-clearing device to work better than the nozzle bump
method?
I'm a former Quinly user, still using their VAAPR bed which works great.
The whole reason I went Octo / Continuous Print is because I had software /
firmware issues I could never get resolved. Well, that was the primary
impetus, however also I wanted to control the clearing script, not pay
monthly fees, learn more, gain independence, etc.  They were a great start
to automated, because at the beginning there was no way I would be ready to
program the printer clearing, etc.
I'm now looking for your awesome clearing script! Just need to find your
article again.
Cheers and thank you again for getting me going on diagnostics!

*RogRoger Rasmussen*
Coastmans Scenic Products
(360) 820-3553
17 Hamlet
Port Orford, Oregon 97465
website: www.coastmans.com
modelmaking: www.coastalmodelworks.com
email: ***@***.***

**Reply by Pierro55** [reply]
Awesome to hear! :)
I can’t say much about if knocking parts off with the nozzle is better because I never tried it actually. But I can give you my unbiased biased perspective from an engineering point of view:
Distributing the forces created from knocking off the parts with a sophisticated system is better than using the x axis of the printer because it is intentionally not designed for that (that also depends on the rigidity of the printer).
In addition, the sweeper arm knocks off parts in one go compared to multiple cycles of the nozzle method.

You can find my script right here in the Loop GitHub repo, all the way down on the instructions page:

https://github.com/Pierro55/Loop/tree/main/Instructions

Learning Gcode will give you much more freedom and flexibility to understand simple logic in start and end Gcode and in general problem solving. Have fun :))

**Comment by rogiepoo** [comment]
I've shared an item with you:

cooling setup
https://drive.google.com/drive/folders/11YxcuyR1rImbqNKk5ZfasARScIJbr3IT?usp=sharing&invite=CK-VguAB&ts=63fe3685

It's not an attachment -- it's stored online. To open this item, just click  
the link above.

Roger sharing with you photos of my 115V fan relay cooling setup. Cheers!   
rr

**Reply by Pierro55** [reply]
Awesome, thanks for sharing. Looks like a good idea :)

**Comment by rogiepoo** [comment]
Pierre,
You may recall I shared my fan cooling system. One caveat is that the fan
shuts off once the build plate begins moving again. I think it would be
highly beneficial if the fan were to remain on during the bed clearing
event.
     I am using the nozzle and path clearing system, for now, as many of my
prints are shallow in Z height, only ~ 2-3mm. I like that I can run the
nozzle path only 0.5mm from the bed without error.
    Frequently, during the bed clearing event, lightweight parts may not
fall from the bed. Also, the primeline, necessary for preserving the detail
in many of the parts, also does not clear the bed. The result is that they
might adhere to a semi-hot nozzle, e.g. 170C, during bed leveling. The
result: a sticky mess.
    Now this next part is beyond my skills, but I believe it is possible.
On the Einsy Rambo board, there could be a pin or other available circuit
output, potentially manipulate-able by virtue of G-code (end G-code
perhaps?), with which a current signal could be induced. This in effect
would negate the need for a mechanical fan "switch" actuated by the
position of the bed. Then, the timing and duration of the fan event could
be accessed through the G-code or perhaps CPQ.
   With additional fans, I am hoping to 1) improve reliability by literally
"blowing" the finished parts right off the bed (with nozzle prompting, of
course) and 2) further reduce the time required for bed cooling.
Your thoughts please, if any.
Yours truly,
Roger


*RogRoger Rasmussen*
Coastmans Scenic Products
(360) 820-3553
17 Hamlet
Port Orford, Oregon 97465
website: www.coastmans.com
modelmaking: www.coastalmodelworks.com
email: ***@***.***

**Reply by Pierro55** [reply]
Hey Roger,

that sounds like a nice idea :)

I have 3 potential solutions on my mind and I will list them from easy to more difficult:

1) You could simply use a different type of switch, for example a latching switch which once pressed holds its position and closes the circuit. Press it again to break the circuit. 

2) You could use a microcontroller (Arduino Nano or similar) which recognises that the switch is pressed, turns on the fan and turns it off when the switch is pressed again. You would need basic programming skills and knowledge in electronics. And if you are working with a mains powered switch please be extremely careful as this can be very dangerous.


3) You could dig through the Prusa Marlin firmware and do what you described in your post. It is definitely possible but you have to do some reading on how to change pins, how to compile the firmware and other stuff. I for myself don’t reallly like working with Marlin firmware as it is not quite user friendly because you have to dig through the complete source code. Nowadays it is somewhat easier but definitely not as easy as other firmware’s (Duet RepRap firmware for instance).

As general advice when controlling mains devices with a microcontroller (Einsy board or any other microcontroller), it is recommended to use a AC-DC SSR with proper current rating. Make sure to not buy the cheapest one, these can be extremely dangerous. A mid class SSR should cost around 10-20€ (here in Germany). 
And I can’t stress this enough, be careful around mains voltage.


One last thing: I would first manually test if the fan can blow off a freshly printed part before you do all the work.

**Comment by rogiepoo** [comment]
Pierre,
Thank you so kindly! I very much appreciate your advice. So hard for me to
find that kind of help here.  Is everyone in Germany as friendly as you?  :)
I do like idea #1 of course in light of what I'd be getting into with #2
and 3.  #3 seems to have the greatest potential as far as unlocking the
true potential of peripherals, and that does pique my interest. But I do
need to press on with production of my things.
I will need to pick apart your advice on the AC-DC SSR; solid-state relay,
I believe?  Yes, I see now, a cheap one could really be a liability, with
115V going through this, here in US.  I will replace it as I am sure this
one is the inexpensive sort you speak of.
I will do a blow test!  I do plan to have the nozzle doing the initial
release, as I do not think air of reasonable velocity in and of itself
could really do the job.
Thanks again!

*RogRoger Rasmussen*
Coastmans Scenic Products
(360) 820-3553
17 Hamlet
Port Orford, Oregon 97465
website: www.coastmans.com
modelmaking: www.coastalmodelworks.com
email: ***@***.***

**Reply by Pierro55** [reply]
I am glad to hear that I can help!
I‘ve learned so much through the open source community (and still are) and I am really happy that I can give it back to anyone who needs it :)

I wish you all the best with your project! 
Have a nice weekend :)

**Comment by rogiepoo** [comment]
I knew there had to be a reason they named this relay "OONO"  aka OH NO!
:)

*RogRoger Rasmussen*
Coastmans Scenic Products
(360) 820-3553
17 Hamlet
Port Orford, Oregon 97465
website: www.coastmans.com
modelmaking: www.coastalmodelworks.com
email: ***@***.***

**Comment by rogiepoo** [comment]
Hi Pierre,
I have been reading and now going back and forth on whether to find a
push-on push-off switch versus changing the relay that is connected to the
115V fan. I've looked at many switches and relays but nothing seems to be
right.  Do you think there might be someone on Git to help with this?  I
should check with Scott.
    This company is in China. I have the F-1050 relay but it's not the
latching type. I'm looking at the F-1030 but not sure if it's going to
work. I wish I was an electronics pro.  I may post for help on SparkFun.
Cheers,
R
F-1050
<https://czh-labs.com/products/wired-dc-ac-3amp-isolated-solid-state-relay-module-ssr-in-3-32vdc-out-90-250vac>
F-1030
<https://czh-labs.com/products/dpst-1no-1nc-8amp-latching-relay-module-version-dc5v>


*RogRoger Rasmussen*
Coastmans Scenic Products
(360) 820-3553
17 Hamlet
Port Orford, Oregon 97465
website: www.coastmans.com
modelmaking: www.coastalmodelworks.com
email: ***@***.***

**Reply by Pierro55** [reply]
Hm, the relays look fishy and unsafe. They should at least have a CE marking (which states that they meet safety regulations etc. but Chinese brands with cheap products often use them without really testing their devices so be careful).
I don't know if this helps because I don't have a VPN but this type of switch should probably do the trick:
https://www.amazon.com/-/de/dp/B088D2LQ19/ref=sr_1_4?__mk_de_DE=%C3%85M%C3%85%C5%BD%C3%95%C3%91&crid=2YGL6Z8NH1XYQ&keywords=latching+switch+110v&qid=1679767438&sprefix=latching+switch+110%2Caps%2C172&sr=8-4

And don't worry about not being an electronics pro, if you keep digging, you will eventually grow a lot of knowledge you can lean on in future projects.

**Comment by rogiepoo** [comment]
Hey thanks, I will check that out immediately.  Yes, I agree, having 115V
flowing in something that doesn't meet specs, that is kinda sketchy.  Those
OH no guys have so many cool things on their site. I've been in
communication with someone there. I really think the F-1030 is the one, but
preferably I would find the exact same thing somewhere here in states or
anywhere with CE rating as you say. Thank you again!
R

*RogRoger Rasmussen*
Coastmans Scenic Products
(360) 820-3553
17 Hamlet
Port Orford, Oregon 97465
website: www.coastmans.com
modelmaking: www.coastalmodelworks.com
email: ***@***.***

**Comment by rogiepoo** [comment]
Ah, for some reason the link tried to open a git hub page instead of
Amazon. If you can send a photo or anything that would be great too.
Thanks

*RogRoger Rasmussen*
Coastmans Scenic Products
(360) 820-3553
17 Hamlet
Port Orford, Oregon 97465
website: www.coastmans.com
modelmaking: www.coastalmodelworks.com
email: ***@***.***

**Reply by Pierro55** [reply]
Just deleted the GitHub link function. The link from my post should probably work now.

**Comment by rogiepoo** [comment]
Hi, I do want to see this, but still takes me to a 404 GitHub page for some
reason. Dang! :( :)

*RogRoger Rasmussen*
Coastmans Scenic Products
(360) 820-3553
17 Hamlet
Port Orford, Oregon 97465
website: www.coastmans.com
modelmaking: www.coastalmodelworks.com
email: ***@***.***

**Reply by Pierro55** [reply]
Hm, try searching for „latching switch 110V“  on Amazon and keep looking for something like this:
![D676B5EE-503B-41E2-94DD-D19DC8819CDA](https://user-images.githubusercontent.com/57869093/227801112-a7ce6ecd-481f-45ba-92e9-340d2c2cf9f0.png)
